universal-application-tool-0.0.1/app/modules/SecurityModule.java

```diff
@@ -145,6 +149,45 @@ protected OidcClient provideIDCSClient(
     return client;
   }
 
+  /**
+   * Creates a singleton object of OidcClient configured for LoginRadius and initializes it on
+   * startup.
+   */
+  @Provides
+  @Nullable
+  @Singleton
+  @LoginRadiusSamlClient
+  protected SAML2Client provideLoginRadiusClient(
+      ProfileFactory profileFactory, Provider<UserRepository> applicantRepositoryProvider) {
+    if (!this.configuration.hasPath("login_radius.keystore_password")
+        || !this.configuration.hasPath("login_radius.private_key_password")
+        || !this.configuration.hasPath("login_radius.api_key")) {
+      return null;
+    }
+
+    String metadataResourceUrl =
+        String.format(
+            "%s?apikey=%s&appName=%s",
+            this.configuration.getString("login_radius.metadata_uri"),
+            this.configuration.getString("login_radius.api_key"),
+            this.configuration.getString("login_radius.saml_app_name"));
+    SAML2Configuration config = new SAML2Configuration();
+    config.setKeystoreResourceFilepath(this.configuration.getString("login_radius.keystore_name"));
+    config.setKeystorePassword(this.configuration.getString("login_radius.keystore_password"));
+    config.setPrivateKeyPassword(this.configuration.getString("login_radius.private_key_password"));
+    config.setIdentityProviderMetadataResourceUrl(metadataResourceUrl);
+    SAML2Client client = new SAML2Client(config);
```

Comment:
does this client need to set the name here to set up the callback correctly? i think the AD client had to do that not sure if this needs to as well

Reply:
The callback is working, so I don't think so? the idcs client also doesn't have a name set.